### DISCUSSION TITLE
Externally triggered DAG's fails when there is a queue spike

### DISCUSSION THREAD

**Discussion by tomerav0** [apache/airflow#24097] [Q&A]
### Apache Airflow version

2.1.2

### Operating System

Google container-Optimized OS

### Versions of Apache Airflow Providers

google-cloud-bigquery
pandas
gcsfs
google-cloud-storage
google-cloud-logging
google-cloud-firestore
google-oauth2-tool
google-auth
psutil
boto3

### Deployment

Composer

### Deployment details

composer-1.17.1 with Google container-Optimized OS running on GKE 1.19.13-gke.1200 using 3 workers of GCP n2-highcpu-8 machines.

SQL server: db-n1-standard-2 (2 vCPU, 7.5 GB memory)

### What happened

When externally triggering many DAG's many of them fails right away with the error of:

`ERROR - Executor reports task instance <TaskInstance: xxxx [queued]> finished (failed) although the task says its queued. (Info: None) Was the task killed externally? `

Another error showing:

`Adopted tasks were still pending after 0:10:00, assuming they never made it to celery and clearing`

The number of failing DAG's is correlated to the size of the spike of the queued tasks a screenshot of the system:

https://imgur.com/a/mFvZRwg

### What you expected to happen

The DAG's should not fail right away 

### How to reproduce

To reproduce the DAG's create a 1000 DAG's dynamically a simple DAG code as this:

```
from airflow import DAG
from airflow.operators.python_operator import PythonOperator
from datetime import datetime, timezone
import time

NUMBER_OF_DAGS = 1000

def run():
    for sleep_minute in range(1, 2):
        print('sleep step:', sleep_minute)
        time.sleep(sleep_minute * 60)

def create_new_dag(name):

    default_args = {
        "depends_on_past": False,
        "email_on_failure": True,
        "email_on_retry": False,
        "retries": 2,
        "start_date": datetime(2019, 12, 5),
    }

    return DAG(
        name,
        default_args=default_args,
        schedule_interval=None,
        catchup=False,
        is_paused_upon_creation=False,
        concurrency = 1
    )

def set_process_python_operator(dag, task_id):
     
     return PythonOperator(
        task_id=task_id,
        provide_context=True,
        python_callable=run,
        dag=dag)

def map_dags():

    for dag_id in range(1, NUMBER_OF_DAGS):

        dag_name = "v2_{}_dag".format(dag_id)
        task_name = "v2_{}_task".format(dag_id)

        dag = create_new_dag(dag_name)

        globals()[dag_name] = dag
       
        set_process_python_operator(dag, task_name)

map_dags()
```

Externally triggering 200 DAG's at the same time using the API causes the above errors. 

### Anything else

I have tried to tune a lot of the scheduler parameters but we no success. 

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by ephraimbuddy** [comment]
Have you tested this in 2.1.4, that's the latest version of airflow? The first error could happen if you have database connection errors. For the second error, the cleared tasks should be resent.

**Comment by tomerav0** [comment]
> Have you tested this in 2.1.4, that's the latest version of airflow? The first error could happen if you have database connection errors. For the second error, the cleared tasks should be resent.

I dont have access to v2.1.4 right now in my environment. 
How can I check if its database connection errors?

**Comment by ephraimbuddy** [comment]
Look into the scheduler logs for any logs before the first error above

**Comment by tomerav0** [comment]
> Look into the scheduler logs for any logs before the first error above

There is no errors. The task is adopted in PENDING state and after 2 seconds got the fail error:

```
2021-10-10T02:19:34 - <TaskInstance: task_xxxxx [queued]> in state PENDING

2021-10-10T02:19:36 - {scheduler_job.py:1218} INFO - Executor reports execution of task_xxxxx exited with status failed for try_number 1

2021-10-10T02:19:36 - {scheduler_job.py:1254} ERROR - Executor reports task instance <TaskInstance:task_xxxxx [queued]> finished (failed) although the task says its queued. (Info: None) Was the task killed externally?"
```
